sotodlib/site_pipeline/analyze_bright_ptsrc.py

```diff
@@ -63,71 +48,77 @@ def get_xieta_src_centered(ctime, az, el, data, sso_name, threshold=5):
 
     csl = so3g.proj.CelestialSightLine.az_el(ctime, az, el, weather="typical")
     q_bore = csl.Q
-    
-    ### Signal strength criterion - a chance to flag detectors (should be extra rare)
+
+    # Signal strength criterion - a chance to flag detectors (should be extra
+    # rare)
     peaks, _ = scipy.signal.find_peaks(data, height=threshold * np.std(data))
     if not list(peaks):
         return
 
-    ## The mean planet position is used
-    dt = datetime.fromtimestamp(np.mean(ctime))
-    dt_str = dt.strftime("%Y/%m/%d %H:%M:%S")
-    obj = getattr(ephem, sso_name)()
-    obj.compute(dt_str)
-    ra_obj = obj.ra
-    dec_obj = obj.dec
-    q_obj = quat.rotation_lonlat(ra_obj, dec_obj)
-
-    ### rotate coord system around max point
+    # Need the planet position at the peak
     src_idx = np.where(data == np.max(data[peaks]))[0][0]
-    q_bs_t = coords.quat.quat(*np.array(q_bore)[src_idx])
-    xi0, eta0, _ = quat.decompose_xieta(~q_bs_t * q_obj)
-    q_det = quat.rotation_xieta(xi0, eta0)
-    q_total = ~q_det * ~q_bore * q_obj
+    d1_unix = ctime[src_idx]
+    planet = planets.SlowSource.for_named_source(sso_name, d1_unix * 1.)
+    ra0, dec0 = planet.pos(d1_unix)
+    q_obj = so3g.proj.quat.rotation_lonlat(ra0, dec0)
+
+    # rotate coord system around max point
+    q_bs_t = coords.quat.quat(*np.array(~q_bore * q_obj)[src_idx])
+    q_total = ~q_bs_t * ~q_bore * q_obj
```

Comment:
What is q_total, and later (xi, eta), supposed to represent?

The docstring or the comments here need to be more explicit, I think, otherwise we can't check the math.

Reply:
Both q_total and q_det project the source's center on the focal plane; q_total projects on the boresight and q_det on the detector in question, respectively. The pairs (xi, eta) and (xi, eta0) are estimated from `decompose_xieta` of these 2 quaternions. No information on the gamma angle is required.